src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO.Pipelines;
+using System.Net.WebSockets;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Sockets.Client
+{
+    public class WebSocketsTransport : ITransport
+    {
+        private ClientWebSocket _webSocket = new ClientWebSocket();
+        private IChannelConnection<Message> _application;
+        private CancellationToken _cancellationToken = new CancellationToken();
+
+        public Task Running { get; private set; }
+                                                
+        public async Task StartAsync(Uri url, IChannelConnection<Message> application)  
+        {
+            if (url == null)
+            {
+                throw new ArgumentNullException(nameof(url));
+            }
+            if (application == null)
+            {
+                throw new ArgumentNullException(nameof(application));
+            }
+            _application = application;
+            var webSocketUrl = url.ToString();
```

Comment:
Consider not converting to String. WebSocketClient takes Uri. (I also am not convinced that `.ToString()` is the right way to get a string version of the Uri)